tests/api_tests/nodes/test_views.py

```diff
@@ -605,6 +605,81 @@ def test_partial_update_private_project_logged_in_non_contributor(self):
         res = self.app.patch_json(self.private_url, {'title': self.new_title}, auth=self.basic_auth_two, expect_errors=True)
         assert_equal(res.status_code, 403)
 
+class TestNodeDelete(ApiTestCase):
+
+    def setUp(self):
+        ApiTestCase.setUp(self)
+        self.user = UserFactory.build()
+        self.user.set_password('password')
+        self.user.save()
+        self.basic_auth = (self.user.username, 'password')
+
+        self.project = ProjectFactory(creator=self.user, is_public=False)
+        self.private_url = '/{}nodes/{}/'.format(API_BASE, self.project._id)
+
+        self.user_two = UserFactory.build()
+        self.user_two.set_password('password')
+        self.user_two.save()
+        self.basic_auth_two = (self.user_two.username, 'password')
+
+        self.public_project = ProjectFactory(is_public=True, creator=self.user)
+        self.public_url = '/{}nodes/{}/'.format(API_BASE, self.public_project._id)
+
+        self.fake_url = '/{}nodes/{}/'.format(API_BASE, '12345')
+
+    def test_deletes_public_node_logged_out(self):
+        res = self.app.delete(self.public_url, expect_errors=True)
+        # This is 403 instead of 401 because basic authentication is only for unit tests and, in order to keep from
+        # presenting a basic authentication dialog box in the front end. We may change this as we understand CAS
+        # a little better
+        assert_equal(res.status_code, 403)
+
+    def test_deletes_public_node_logged_in(self):
+        res = self.app.delete(self.public_url, auth=self.basic_auth_two, expect_errors=True)
+        assert_equal(res.status_code, 403)
+        assert_equal(self.public_project.is_deleted, False)
+
+        res = self.app.delete(self.public_url, auth=self.basic_auth, expect_errors=True)
+        assert_equal(res.status_code, 400)
```

Comment:
This route returning a 400 (or any 4xx) feels wrong to me. My understanding of the 4xx series codes is that they should be sent only when there is an error made by the client. As this is the intended flow for a node deletion, returning a client error doesn't make sense.

After careful review of the applicable RFCs, I propose we use [`202 ACCEPTED`](https://tools.ietf.org/html/rfc7231#section-6.3.3) instead. 

> The 202 (Accepted) status code indicates that the request has been
   accepted for processing, but the processing has not been completed.
   The request might or might not eventually be acted upon, as it might
   be disallowed when processing actually takes place.  There is no
   facility in HTTP for re-sending a status code from an asynchronous
   operation.

> The 202 response is intentionally noncommittal.  Its purpose is to
   allow a server to accept a request for some other process (perhaps a
   batch-oriented process that is only run once per day) without
   requiring that the user agent's connection to the server persist
   until the process is completed.  The representation sent with this
   response ought to describe the request's current status and point to
   (or embed) a status monitor that can provide the user with an
   estimate of when the request will be fulfilled.

By returning a 202, we're effectively saying "OK, cool. You want that node deleted, and I hear you loud and clear. I'll take care of it, as soon as I'm ready."

Since the spec also allows for a status monitor to be returned. The mechanism for that is undefined, but I assume it would be either a `Location: ` header, or in the body of the response. Our API could then specify that this URL must be hit with a `DELETE` request before the node will actually be deleted.

This seems to satisfy the desire for a two-step process for deleting a node, while avoiding a response indicating an error.